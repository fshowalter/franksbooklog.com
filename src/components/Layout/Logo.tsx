import { ccn } from "src/utils/concatClassNames";

export function Logo({ className }: { className: string }) {
  return (
    <div className={ccn("flex flex-col", className)}>
      <div
        className="whitespace-nowrap font-normal leading-8"
        style={{ fontSize: "1.5625rem" }}
      >
        <a href="/">Frank&apos;s Book Log</a>
      </div>
      <p className={"w-full pl-px text-sm italic leading-4 opacity-85"}>
        Literature is a relative term.
      </p>
    </div>
  );
}
