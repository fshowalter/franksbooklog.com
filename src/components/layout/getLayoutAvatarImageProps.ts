import { getAvatarImageProps } from "~/api/avatars";

import { AvatarImageConfig } from "./Backdrop";

/**
 * Fetches backdrop image properties for the layout component.
 * @param slug - The slug identifier for the backdrop image
 * @returns Backdrop image properties configured for layout use
 */
export async function getLayoutAvatarImageProps(slug: string) {
  return getAvatarImageProps(slug, AvatarImageConfig);
}
