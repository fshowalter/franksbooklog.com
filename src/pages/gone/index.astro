---
import AstroPageShell from "~/astro/AstroPageShell.astro";
import { getLayoutBackdropImageProps } from "~/components/layout/getLayoutBackdropImageProps";
import { Layout } from "~/components/layout/Layout";

const backdropImageProps = await getLayoutBackdropImageProps("gone");
const deck = '"This is not an exit."';
---

<AstroPageShell
  canonical={false}
  meta={{ description: deck }}
  openGraph={false}
  title="410: Gone"
>
  <Layout
    backdrop={{
      backdropImageProps,
      centerText: true,
      deck,
      size: "large",
      title: "410 Gone",
    }}
    className="bg-subtle"
  >
    <span></span>
  </Layout>
</AstroPageShell>
