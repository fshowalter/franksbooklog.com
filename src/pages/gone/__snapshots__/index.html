<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta content="width=device-width" name="viewport" />
    <link href="/favicon.ico" rel="icon" sizes="48x48" />
    <link href="/favicon.svg" rel="icon" sizes="any" type="image/svg+xml" />
    <link href="/apple-touch-icon.png" rel="apple-touch-icon" />
    <link href="/site.webmanifest" rel="manifest" />
    <link href="/sitemap-index.xml" rel="sitemap" />
    <meta
      content="&#34;Forget it, Jake. It's Chinatown.&#34;"
      name="description"
    />
    <meta
      content="#f2f0e8"
      media="(prefers-color-scheme: light)"
      name="theme-color"
    />
    <meta
      content="#484745"
      media="(prefers-color-scheme: dark)"
      name="theme-color"
    />
    <link
      href="https://www.franksbooklog.com/feed.xml"
      rel="alternate"
      title="Frank's Book Log"
      type="application/rss+xml"
    />
    <title>410: Gone - Frank&#39;s Book Log</title>
  </head>
  <body>
    <div>
      <a
        class="absolute left-1/2 top-0.5 z-50 mx-auto bg-subtle px-6 py-2 text-center text-accent [transform:translate(-50%,calc(-100%_-_2px))] focus:[transform:translate(-50%,0%)]"
        href="#content"
        >Skip to content</a
      >
      <div class="flex min-h-full w-full flex-col bg-default">
        <header
          class="z-20 flex w-full items-center justify-between px-container py-4 tablet:p-6 desktop:inset-x-0 desktop:z-40 desktop:flex-row desktop:flex-wrap desktop:px-16 desktop:py-8 desktop:text-left"
          style="position: absolute"
        >
          <div class="flex flex-col">
            <div
              class="whitespace-nowrap font-normal leading-8"
              style="font-size: 1.5625rem"
            >
              <a href="/">Frank&#x27;s Book Log</a>
            </div>
            <p class="w-full pl-px text-sm italic leading-4 opacity-85">
              Literature is a relative term.
            </p>
          </div>
          <div class="flex items-center">
            <nav class="hidden w-full desktop:block desktop:w-auto">
              <ul class="flex flex-wrap justify-start gap-x-6 text-xl">
                <li class="block whitespace-nowrap tracking-serif-wide">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                    href="/"
                    >Home</a
                  >
                </li>
                <li class="block whitespace-nowrap tracking-serif-wide">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                    href="/how-i-grade/"
                    >How I Grade</a
                  >
                </li>
                <li class="block whitespace-nowrap tracking-serif-wide">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                    href="/reviews/"
                    >Reviews</a
                  >
                </li>
                <li class="block whitespace-nowrap tracking-serif-wide">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                    href="/authors/"
                    >Authors</a
                  >
                </li>
                <li class="block whitespace-nowrap tracking-serif-wide">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:text-accent"
                    href="/readings/"
                    >Reading Log</a
                  >
                </li>
              </ul>
            </nav>
            <div class="search-button z-[1000]">
              <button
                aria-keyshortcuts="Control+K"
                aria-label="Search"
                class="flex items-center justify-between overflow-hidden text-sm leading-6 ring-default desktop:ml-6"
                data-open-modal="true"
                disabled=""
                title="Search: Control+K"
                type="button"
              >
                <svg
                  class="size-7"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m21 21-5.197-5.197m0 0A7.5 7.5 0 1 0 5.196 5.196a7.5 7.5 0 0 0 10.607 10.607Z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  ></path>
                </svg>
              </button>
            </div>
            <input class="hidden" id="mobile-nav" type="checkbox" /><label
              class="hamburger-icon desktop:hidden"
              for="mobile-nav"
              ><span
                class="hamburger-icon-bars"
                style="background-color: var(--fg-default)"
              ></span
            ></label>
            <ul
              class="hamburger-menu flex flex-col items-start gap-y-5 text-left text-inverse desktop:hidden"
            >
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/">Home</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/how-i-grade/">How I Grade</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/reviews/">Reviews</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/authors/">Authors</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/readings/">Reading Log</a>
                <ol class="mt-4">
                  <li
                    class="mb-2 font-sans text-xs uppercase tracking-wide text-inverse-subtle"
                  >
                    <a href="/readings/stats/">Stats</a>
                  </li>
                </ol>
              </li>
            </ul>
          </div>
        </header>
        <main class="grow" id="content">
          <article>
            <header
              class="min-h-[clamp(340px,50vh,1350px)] relative flex w-full flex-col content-start items-center justify-end gap-6 bg-canvas pb-8 pt-40 tablet:pb-10 tablet:pt-40 desktop:pb-16 desktop:pt-40"
            >
              <div
                class="z-10 mx-auto flex w-full px-container text-center tablet:max-w-unset tablet:px-0 flex-col"
              >
                <h1
                  class="font-sans text-3xl font-bold uppercase tracking-widest tablet:text-4xl tablet-landscape:text-5xl desktop:text-7xl"
                >
                  410 Gone
                </h1>
                <p class="mt-1 text-base desktop:my-4 desktop:text-xl">
                  &quot;Forget it, Jake. It&#x27;s Chinatown.&quot;
                </p>
              </div>
            </header>
            <section class="flex flex-col items-center pb-32 pt-16">
              <div class="px-container">
                <div
                  class="rendered-markdown tracking-0.3px text-md/7 tablet:text-xl/8 max-w-prose"
                >
                  <p>Whatever you’re looking for, it’s not coming back.</p>
                </div>
              </div>
            </section>
          </article>
          <div
            class="flex w-full flex-col items-center gap-y-12 bg-subtle pb-32 pt-16 tablet:pt-8 desktop:gap-y-24"
            data-pagefind-ignore="true"
          >
            <nav
              class="mx-auto w-full max-w-screen-max bg-subtle tablet:max-w-popout desktop:max-w-screen-max desktop:px-container"
              data-page-find-ignore="true"
            >
              <div class="px-container tablet:px-0">
                <h2
                  class="py-10 font-sans text-xs font-semibold uppercase tracking-wide text-subtle"
                >
                  Recent<!-- -->
                  <a class="text-accent" href="/reviews/">Reviews</a>
                </h2>
              </div>
              <ul
                class="flex flex-wrap justify-center gap-x-[4%] gap-y-[6vw] px-[4%] tablet:gap-x-[3%] tablet:px-0 desktop:justify-between desktop:gap-x-[2%]"
              >
                <li
                  class="relative flex w-[48%] max-w-[248px] flex-col items-center border-default bg-cover-back has-[a:hover]:shadow-hover min-[600px]:w-[30.66666667%] tablet:w-[31.33333333%] desktop:w-[14.16666667%]"
                >
                  <img
                    src="/_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fghost-story-by-peter-straub.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif"
                    srcset="
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fghost-story-by-peter-straub.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;h=93&amp;q=80&amp;f=avif    62w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fghost-story-by-peter-straub.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;h=186&amp;q=80&amp;f=avif 124w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fghost-story-by-peter-straub.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif 248w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fghost-story-by-peter-straub.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;h=744&amp;q=80&amp;f=avif 496w
                    "
                    height="372"
                    sizes="(min-width: 1800px) 218px, (min-width: 1280px) calc(11.8vw + 8px), (min-width: 820px) 216px, (min-width: 600px) calc(23.5vw + 28px), calc(41.43vw + 8px)"
                    width="248"
                    alt=""
                    decoding="async"
                    loading="lazy"
                    style="aspect-ratio: 0.66666667"
                  />
                  <div
                    class="flex w-full grow flex-col items-center bg-default px-[8%] pb-8 pt-3 has-[a:hover]:bg-stripe desktop:pl-[8.5%] desktop:pr-[10%]"
                  >
                    <div class="text-center text-md font-medium leading-6">
                      <a
                        class="z-10 inline-block text-accent decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 after:absolute after:left-0 after:top-0 after:size-full after:opacity-0 hover:before:opacity-0"
                        href="/reviews/ghost-story-by-peter-straub/"
                        rel="canonical"
                        >Ghost Story</a
                      >
                    </div>
                    <div
                      class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                    >
                      1979<!-- -->
                      |<!-- -->
                      <span class="whitespace-nowrap">Novel</span>
                    </div>
                    <p class="text-center text-base text-subtle">
                      <span class="font-light">by</span>
                      <!-- -->Peter Straub
                    </p>
                    <picture
                      ><source
                        media="(prefers-color-scheme: dark)"
                        srcset="/svg/4-stars-dark.svg" />
                      <img
                        alt="B: 4 stars (out of 5)"
                        class="mt-2"
                        height="18"
                        src="/svg/4-stars.svg"
                        width="90"
                    /></picture>
                  </div>
                </li>
                <li
                  class="relative flex w-[48%] max-w-[248px] flex-col items-center border-default bg-cover-back has-[a:hover]:shadow-hover min-[600px]:w-[30.66666667%] tablet:w-[31.33333333%] desktop:w-[14.16666667%]"
                >
                  <img
                    src="/_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fgoth-by-lol-tolhurst.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif"
                    srcset="
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fgoth-by-lol-tolhurst.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;h=93&amp;q=80&amp;f=avif    62w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fgoth-by-lol-tolhurst.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;h=186&amp;q=80&amp;f=avif 124w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fgoth-by-lol-tolhurst.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif 248w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fgoth-by-lol-tolhurst.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;h=744&amp;q=80&amp;f=avif 496w
                    "
                    height="372"
                    sizes="(min-width: 1800px) 218px, (min-width: 1280px) calc(11.8vw + 8px), (min-width: 820px) 216px, (min-width: 600px) calc(23.5vw + 28px), calc(41.43vw + 8px)"
                    width="248"
                    alt=""
                    decoding="async"
                    loading="lazy"
                    style="aspect-ratio: 0.66666667"
                  />
                  <div
                    class="flex w-full grow flex-col items-center bg-default px-[8%] pb-8 pt-3 has-[a:hover]:bg-stripe desktop:pl-[8.5%] desktop:pr-[10%]"
                  >
                    <div class="text-center text-md font-medium leading-6">
                      <a
                        class="z-10 inline-block text-accent decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 after:absolute after:left-0 after:top-0 after:size-full after:opacity-0 hover:before:opacity-0"
                        href="/reviews/goth-by-lol-tolhurst/"
                        rel="canonical"
                        >Goth</a
                      >
                    </div>
                    <div
                      class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                    >
                      2023<!-- -->
                      |<!-- -->
                      <span class="whitespace-nowrap">Nonfiction</span>
                    </div>
                    <p class="text-center text-base text-subtle">
                      <span class="font-light">by</span>
                      <!-- -->Lol Tolhurst
                    </p>
                    <picture
                      ><source
                        media="(prefers-color-scheme: dark)"
                        srcset="/svg/3-stars-dark.svg" />
                      <img
                        alt="C: 3 stars (out of 5)"
                        class="mt-2"
                        height="18"
                        src="/svg/3-stars.svg"
                        width="90"
                    /></picture>
                  </div>
                </li>
                <li
                  class="relative flex w-[48%] max-w-[248px] flex-col items-center border-default bg-cover-back has-[a:hover]:shadow-hover min-[600px]:w-[30.66666667%] tablet:w-[31.33333333%] desktop:w-[14.16666667%]"
                >
                  <img
                    src="/_image/?href=%2Fcontent%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif"
                    srcset="
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;h=93&amp;q=80&amp;f=avif    62w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;h=186&amp;q=80&amp;f=avif 124w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif 248w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Frecessional-by-david-mamet.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;h=744&amp;q=80&amp;f=avif 496w
                    "
                    height="372"
                    sizes="(min-width: 1800px) 218px, (min-width: 1280px) calc(11.8vw + 8px), (min-width: 820px) 216px, (min-width: 600px) calc(23.5vw + 28px), calc(41.43vw + 8px)"
                    width="248"
                    alt=""
                    decoding="async"
                    loading="lazy"
                    style="aspect-ratio: 0.66666667"
                  />
                  <div
                    class="flex w-full grow flex-col items-center bg-default px-[8%] pb-8 pt-3 has-[a:hover]:bg-stripe desktop:pl-[8.5%] desktop:pr-[10%]"
                  >
                    <div class="text-center text-md font-medium leading-6">
                      <a
                        class="z-10 inline-block text-accent decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 after:absolute after:left-0 after:top-0 after:size-full after:opacity-0 hover:before:opacity-0"
                        href="/reviews/recessional-by-david-mamet/"
                        rel="canonical"
                        >Recessional</a
                      >
                    </div>
                    <div
                      class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                    >
                      2022<!-- -->
                      |<!-- -->
                      <span class="whitespace-nowrap">Nonfiction</span>
                    </div>
                    <p class="text-center text-base text-subtle">
                      <span class="font-light">by</span>
                      <!-- -->David Mamet
                    </p>
                    <picture
                      ><source
                        media="(prefers-color-scheme: dark)"
                        srcset="/svg/2-stars-dark.svg" />
                      <img
                        alt="D: 2 stars (out of 5)"
                        class="mt-2"
                        height="18"
                        src="/svg/2-stars.svg"
                        width="90"
                    /></picture>
                  </div>
                </li>
                <li
                  class="relative flex w-[48%] max-w-[248px] flex-col items-center border-default bg-cover-back has-[a:hover]:shadow-hover min-[600px]:w-[30.66666667%] tablet:w-[31.33333333%] desktop:w-[14.16666667%]"
                >
                  <img
                    src="/_image/?href=%2Fcontent%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif"
                    srcset="
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;h=93&amp;q=80&amp;f=avif    62w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;h=186&amp;q=80&amp;f=avif 124w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif 248w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Ffast-forward-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;h=744&amp;q=80&amp;f=avif 496w
                    "
                    height="372"
                    sizes="(min-width: 1800px) 218px, (min-width: 1280px) calc(11.8vw + 8px), (min-width: 820px) 216px, (min-width: 600px) calc(23.5vw + 28px), calc(41.43vw + 8px)"
                    width="248"
                    alt=""
                    decoding="async"
                    loading="lazy"
                    style="aspect-ratio: 0.66666667"
                  />
                  <div
                    class="flex w-full grow flex-col items-center bg-default px-[8%] pb-8 pt-3 has-[a:hover]:bg-stripe desktop:pl-[8.5%] desktop:pr-[10%]"
                  >
                    <div class="text-center text-md font-medium leading-6">
                      <a
                        class="z-10 inline-block text-accent decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 after:absolute after:left-0 after:top-0 after:size-full after:opacity-0 hover:before:opacity-0"
                        href="/reviews/fast-forward-by-stephen-morris/"
                        rel="canonical"
                        >Fast Forward</a
                      >
                    </div>
                    <div
                      class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                    >
                      2020<!-- -->
                      |<!-- -->
                      <span class="whitespace-nowrap">Nonfiction</span>
                    </div>
                    <p class="text-center text-base text-subtle">
                      <span class="font-light">by</span>
                      <!-- -->Stephen Morris
                    </p>
                    <picture
                      ><source
                        media="(prefers-color-scheme: dark)"
                        srcset="/svg/4-half-stars-dark.svg" />
                      <img
                        alt="A-: 4.5 stars (out of 5)"
                        class="mt-2"
                        height="18"
                        src="/svg/4-half-stars.svg"
                        width="90"
                    /></picture>
                  </div>
                </li>
                <li
                  class="relative flex w-[48%] max-w-[248px] flex-col items-center border-default bg-cover-back has-[a:hover]:shadow-hover min-[600px]:w-[30.66666667%] tablet:w-[31.33333333%] desktop:w-[14.16666667%]"
                >
                  <img
                    src="/_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif"
                    srcset="
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;h=93&amp;q=80&amp;f=avif    62w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;h=186&amp;q=80&amp;f=avif 124w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif 248w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Fgothic-by-philip-fracassi.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;h=744&amp;q=80&amp;f=avif 496w
                    "
                    height="372"
                    sizes="(min-width: 1800px) 218px, (min-width: 1280px) calc(11.8vw + 8px), (min-width: 820px) 216px, (min-width: 600px) calc(23.5vw + 28px), calc(41.43vw + 8px)"
                    width="248"
                    alt=""
                    decoding="async"
                    loading="lazy"
                    style="aspect-ratio: 0.66666667"
                  />
                  <div
                    class="flex w-full grow flex-col items-center bg-default px-[8%] pb-8 pt-3 has-[a:hover]:bg-stripe desktop:pl-[8.5%] desktop:pr-[10%]"
                  >
                    <div class="text-center text-md font-medium leading-6">
                      <a
                        class="z-10 inline-block text-accent decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 after:absolute after:left-0 after:top-0 after:size-full after:opacity-0 hover:before:opacity-0"
                        href="/reviews/gothic-by-philip-fracassi/"
                        rel="canonical"
                        >Gothic</a
                      >
                    </div>
                    <div
                      class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                    >
                      2023<!-- -->
                      |<!-- -->
                      <span class="whitespace-nowrap">Novel</span>
                    </div>
                    <p class="text-center text-base text-subtle">
                      <span class="font-light">by</span>
                      <!-- -->Philip Fracassi
                    </p>
                    <picture
                      ><source
                        media="(prefers-color-scheme: dark)"
                        srcset="/svg/4-stars-dark.svg" />
                      <img
                        alt="B+: 4 stars (out of 5)"
                        class="mt-2"
                        height="18"
                        src="/svg/4-stars.svg"
                        width="90"
                    /></picture>
                  </div>
                </li>
                <li
                  class="relative flex w-[48%] max-w-[248px] flex-col items-center border-default bg-cover-back has-[a:hover]:shadow-hover min-[600px]:w-[30.66666667%] tablet:w-[31.33333333%] desktop:w-[14.16666667%]"
                >
                  <img
                    src="/_image/?href=%2Fcontent%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif"
                    srcset="
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=62&amp;h=93&amp;q=80&amp;f=avif    62w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=124&amp;h=186&amp;q=80&amp;f=avif 124w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=248&amp;h=372&amp;q=80&amp;f=avif 248w,
                      /_image/?href=%2Fcontent%2Fassets%2Fcovers%2Frecord-play-pause-by-stephen-morris.png%3ForigWidth%3D500%26origHeight%3D750%26origFormat%3Dpng&amp;w=496&amp;h=744&amp;q=80&amp;f=avif 496w
                    "
                    height="372"
                    sizes="(min-width: 1800px) 218px, (min-width: 1280px) calc(11.8vw + 8px), (min-width: 820px) 216px, (min-width: 600px) calc(23.5vw + 28px), calc(41.43vw + 8px)"
                    width="248"
                    alt=""
                    decoding="async"
                    loading="lazy"
                    style="aspect-ratio: 0.66666667"
                  />
                  <div
                    class="flex w-full grow flex-col items-center bg-default px-[8%] pb-8 pt-3 has-[a:hover]:bg-stripe desktop:pl-[8.5%] desktop:pr-[10%]"
                  >
                    <div class="text-center text-md font-medium leading-6">
                      <a
                        class="z-10 inline-block text-accent decoration-2 underline-offset-4 before:absolute before:inset-x-0 before:top-0 before:aspect-cover before:bg-[#fff] before:opacity-15 after:absolute after:left-0 after:top-0 after:size-full after:opacity-0 hover:before:opacity-0"
                        href="/reviews/record-play-pause-by-stephen-morris/"
                        rel="canonical"
                        >Record Play Pause</a
                      >
                    </div>
                    <div
                      class="py-2 text-center font-sans text-xxs font-light uppercase leading-4 tracking-wide text-subtle"
                    >
                      2019<!-- -->
                      |<!-- -->
                      <span class="whitespace-nowrap">Nonfiction</span>
                    </div>
                    <p class="text-center text-base text-subtle">
                      <span class="font-light">by</span>
                      <!-- -->Stephen Morris
                    </p>
                    <picture
                      ><source
                        media="(prefers-color-scheme: dark)"
                        srcset="/svg/5-stars-dark.svg" />
                      <img
                        alt="A: 5 stars (out of 5)"
                        class="mt-2"
                        height="18"
                        src="/svg/5-stars.svg"
                        width="90"
                    /></picture>
                  </div>
                </li>
              </ul>
            </nav>
          </div>
        </main>
        <footer
          class="flex flex-wrap items-start justify-between gap-[10%] bg-footer px-container py-20 text-inverse tablet:px-12 tablet:pt-10 desktop:p-20"
        >
          <div class="flex flex-col pb-12">
            <div class="flex flex-col">
              <div
                class="whitespace-nowrap font-normal leading-8"
                style="font-size: 1.5625rem"
              >
                <a href="/">Frank&#x27;s Book Log</a>
              </div>
              <p class="w-full pl-px text-sm italic leading-4 opacity-85">
                Literature is a relative term.
              </p>
            </div>
          </div>
          <div
            class="flex flex-col gap-20 pb-20 pt-10 tablet:basis-1/2 tablet:pt-20"
          >
            <ul class="flex w-full flex-col gap-y-10 text-inverse max:w-auto">
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/">Home</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/how-i-grade/"
                  >How I Grade</a
                >
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/reviews/">Reviews</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/authors/">Authors</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/readings/">Reading Log</a>
                <ol class="mt-4">
                  <li
                    class="mb-2 font-sans text-xs uppercase tracking-wide text-inverse-subtle"
                  >
                    <a class="hover:text-inverse" href="/readings/stats/"
                      >Stats</a
                    >
                  </li>
                </ol>
              </li>
            </ul>
            <a
              class="mb-8 w-full max-w-button bg-canvas py-5 text-center font-sans text-xs uppercase tracking-wide text-default hover:bg-inverse hover:text-inverse"
              href="#top"
              >To the top</a
            >
            <p class="w-full font-normal leading-5 text-inverse-subtle">
              All covers used in accordance with the<!-- -->
              <a
                class="text-inherit underline decoration-2 underline-offset-4 hover:bg-default hover:text-default"
                href="http://www.copyright.gov/title17/92chap1.html#107"
                >Fair Use Law.</a
              >
            </p>
          </div>
        </footer>
      </div>
    </div>
    <dialog
      aria-label="Search"
      class="fixed inset-x-4 h-screen w-full rounded-lg border border-default bg-default shadow-lg backdrop:bg-[#000] backdrop:opacity-40 open:flex open:flex-col tablet:mx-auto tablet:mt-16 tablet:h-min tablet:max-h-[min(100vh_-_128px,900px)] tablet:min-h-60 tablet:max-w-[min(100vw_-_64px,36rem)] tablet:rounded-xl desktop:min-h-0"
    >
      <div class="contents" data-dialog-frame>
        <button
          class="px-2 font-mono text-xs font-light hover:bg-inverse hover:text-inverse focus:outline-accent"
          data-close-modal
        >
          ESC
        </button>
        <div class="contents"><div id="pagefind__search"></div></div>
        <div
          class="hidden px-[var(--search-dialog-padding)] py-4 font-sans text-xs font-light tracking-wide text-subtle desktop:block"
        >
          <kbd class="rounded-md bg-default p-2 font-mono text-xs text-subtle"
            >Tab</kbd
          >
          to navigate
          <kbd
            class="ml-4 rounded-md bg-default p-2 font-mono text-xs text-subtle"
            >ESC</kbd
          >
          to close
        </div>
      </div>
    </dialog>
    <script>
      (() => {
        const openBtn = document.querySelector("button[data-open-modal]");
        if (!openBtn) return;
        if (/(Mac|iPhone|iPod|iPad)/i.test(navigator.platform)) {
          openBtn.setAttribute("aria-keyshortcuts", "Meta+K");
          openBtn.setAttribute("title", `Search: ⌘K`);
        }
      })();
    </script>
    <script
      type="module"
      src="/src/layouts/Layout.astro?astro&type=script&index=0&lang.ts"
    ></script>
  </body>
</html>
